pkg/datastore/applicationstore.go

```diff
@@ -37,14 +37,14 @@ func (a *applicationCollection) Factory() Factory {
 
 type ApplicationStore interface {
 	AddApplication(ctx context.Context, app *model.Application) error
-	EnableApplication(ctx context.Context, id string) error
-	DisableApplication(ctx context.Context, id string) error
-	DeleteApplication(ctx context.Context, id string) error
+	EnableApplication(ctx context.Context, w Writer, id string) error
+	DisableApplication(ctx context.Context, w Writer, id string) error
+	DeleteApplication(ctx context.Context, w Writer, id string) error
 	GetApplication(ctx context.Context, id string) (*model.Application, error)
 	ListApplications(ctx context.Context, opts ListOptions) ([]*model.Application, string, error)
-	UpdateApplication(ctx context.Context, id string, updater func(*model.Application) error) error
-	PutApplicationSyncState(ctx context.Context, id string, syncState *model.ApplicationSyncState) error
-	PutApplicationMostRecentDeployment(ctx context.Context, id string, status model.DeploymentStatus, deployment *model.ApplicationDeploymentReference) error
+	UpdateApplication(ctx context.Context, w Writer, id string, updater func(*model.Application) error) error
+	PutApplicationSyncState(ctx context.Context, w Writer, id string, syncState *model.ApplicationSyncState) error
+	PutApplicationMostRecentDeployment(ctx context.Context, w Writer, id string, status model.DeploymentStatus, deployment *model.ApplicationDeploymentReference) error
```

Comment:
I think we should not update the interface of those functions to avoid updating from their callers.
Instead of that, we can decide `operator` or `writer` or whatever inside those functions and pass it to the underlying update function.

Reply:
The same for other stores as well.